src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -4880,7 +4880,7 @@ else if (DataTypes.UNKNOWN_STREAM_LENGTH == dataLength)
                 isShortValue = columnPair.getValue().precision <= DataTypes.SHORT_VARTYPE_MAX_BYTES;
                 isNull = (null == currentObject);
                 if (currentObject instanceof String)
-                    dataLength = isNull ? 0 : (ParameterUtils.HexToBin(currentObject.toString())).length;
+                    dataLength = ParameterUtils.HexToBin(currentObject.toString()).length;
                 else
                     dataLength = isNull ? 0 : ((byte[]) currentObject).length;
```

Comment:
If isNull is checked above, this check can go away too

Reply:
i think this one stays?

Reply:
Also, this method has 2 object instances `currentObject` and `currentObjectStringValue` that are compared to null over and over again, which can be done in the beginning of the method and a boolean instance for each can be used everywhere we need to check for null. SqlVariant has no null value check for `currentObject` that should be handled too..

Reply:
As discussed, we'll keep that change for later